autofit/graphical/factor_graphs/abstract.py

```diff
@@ -160,6 +178,41 @@ def all_variables(self) -> Set[Variable]:
         """
         return self.variables | self.deterministic_variables
 
+    
+    def broadcast_plates(
+            self,
+            plates: Tuple[Plate],
+            value: np.ndarray
+    ) -> np.ndarray:
+        """
+        Extract the indices of a collection of plates then match
+        the shape of the data to that shape.
+
+        Parameters
+        ----------
+        plates
+            Plates representing the dimensions of some factor
+        value
+            A value to broadcast
+
+        Returns
+        -------
+        The value reshaped to match the plates
+        """
+        shift = np.ndim(value) - len(plates)
+        if shift > 1 or shift < 0:
+            raise ValueError("dimensions of value incompatible with passed plates")
+        reduce_axes = tuple(
+            i + shift for i, p in enumerate(plates) if p and p not in self.plates)
```

Comment:
There's no case in the tests where len(self.plates) > 0 meaning most of the functionality here is untested